src/main/scala/leon/comparison/ComparatorClassList.scala

```diff
@@ -0,0 +1,63 @@
+package leon.comparison
+
+import leon.comparison.Utils._
+import leon.purescala.Expressions._
+
+/**
+  * Created by joachimmuth on 02.05.16.
+  *
+  * This method shares similarities with the ComparatorByList.
+  * We keep the idea of comparing a list of expressions (disregarding their order), but now, instead of comparing
+  * two expressions (i.e. tree) we will extract the type of each expression.
+  *
+  * x match {
+  *   case 1 => 'a'
+  *   case 2 => 'b'
+  * }
+  *
+  * ComparatorByList -> {complete match, leaf(a), leaf(b)}
+  * ComparatorByListType -> {node(match), leaf(a), leaf(b)}
+  *
+  * x match {
+  *   case 1 => 'a'
+  *   case 2 => 'c'
+  * }
+  *
+  * ComparatorByList -> similarity 33%
+  * ComparatorByListType -> similarity 66%
+  */
+object ComparatorClassList extends Comparator {
+  val name = "ClassList"
+
+  def compare(expr_corpus: Expr, expr: Expr) = {
+    val listClassesA = collectClass(expr_corpus)
+    val listClassesB = collectClass(expr)
+
+    val similarExpr: Int = pairsOfSimilarExp(listClassesA, listClassesB)
+
+    val score = Utils.matchScore(similarExpr, listClassesA.size, listClassesB.size)
+
+    if (score > 0.0 && ComparisonPhase.debug){
```

Comment:
Could you please look at my commits 101df6a171ade2b08c9046210b96470112149568 and 5bd355753f6c1aefaaa4783d3872b977bfbce73c and 6349eb2130fa7b3ab053a460d176750d453a346e

I'm not sure if I use correctly the Leon debug especially:
- Is it correct to pass add implicit context to function "combinationOfFunDef" ? (the function that call the comparators)
- Is it okay to declare the implicite DebugSection in each comparator or do I must declare it in the main phase?